UPGRADING.md

```diff
@@ -0,0 +1,184 @@
+    or base64 binaries from the JSON response.
+- [`correctClockSkew`](https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/Config.html#correctClockSkew-property)
+  - **v2**: Whether to apply a clock skew correction and retry requests that fail because of an skewed client
+    clock.
+  - **v3**: **Deprecated**. SDK _always_ applies a clock skew correction.
+- [`systemClockOffset`](https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/Config.html#systemClockOffset-property)
+  - **v2**: An offset value in milliseconds to apply to all signing times.
+  - **v3**: No change.
+- [`credentials`](https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/Config.html#credentials-property)
+  - **v2**: The AWS credentials to sign requests with.
+  - **v3**: No change. It can also be an async function that returns a credential.
+    See [v3 reference](https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/interfaces/_aws_sdk_middleware_signing.awsauthinputconfig-1.html#credentials)
+- [`endpointCacheSize`](https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/Config.html#endpointCacheSize-property)
+  - **v2**: The size of the global cache storing endpoints from endpoint discovery operations.
+  - **v3**: Not available. Planned. This option configures endpoint discovery behavior, which is not yet available in v3.
+- [`endpointDiscoveryEnabled`](https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/Config.html#endpointDiscoveryEnabled-property)
+  - **v2**: Whether to call operations with endpoints given by service dynamically.
+  - **v3**: Not available. Planned. This option configures endpoint discovery behavior, which is not yet available in v3.
+- [`hostPrefixEnabled`](https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/Config.html#hostPrefixEnabled-property)
+  - **v2**: Whether to marshal request parameters to the prefix of hostname.
+  - **v3**: **Deprecated**. SDK _always_ injects the hostname prefix when necessary.
+- [`httpOptions`](https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/Config.html#httpOptions-property)
+
+  A set of options to pass to the low-level HTTP request. These options are aggregated differently in V3. You can
+  configure them by supplying a new `requestHandler`. Here's the example of setting http options in Node.js runtime. You
+  can find more in [v3 reference](https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/classes/_aws_sdk_node_http_handler.nodehttphandler-1.html).
+
+  ```javascript
+  const { Agent } = require("https");
+  const { Agent: HttpAgnet } = require("http");
+  const { NodeHttpHandler } = require("@aws-sdk/node-http-handler");
+  const dynamodbClient = new DynamoDBClient({
+    requestHandler: new NodeHttpHandler({
+      httpsAgent: new Agent({
+        /*params*/
+      }),
+      httpAgent: new HttpAgnet({
+        /*params*/
+      }),
+      connectionTimeout: /*number in milliseconds*/
+      socketTimeout: /*number in milliseconds*/
+    }),
+  });
+  ```
```

Comment:
Suggestion: clarify that both httpAgent and httpsAgents are not required. For example:

---

  All v3 requests use HTTPS by default. You only need to provide custom httpsAgent.
  ```javascript
  const { Agent } = require("https");
  const { NodeHttpHandler } = require("@aws-sdk/node-http-handler");

  const dynamodbClient = new DynamoDBClient({
    requestHandler: new NodeHttpHandler({
      httpsAgent: new Agent({
        /*params*/
      }),
      connectionTimeout: /*number in milliseconds*/
      socketTimeout: /*number in milliseconds*/
    }),
  });
  ```

  If you are passing custom endpoint which uses http, then you need to provide httpAgent.
  ```javascript
  const { Agent } = require("http");
  const { NodeHttpHandler } = require("@aws-sdk/node-http-handler");

  const dynamodbClient = new DynamoDBClient({
    requestHandler: new NodeHttpHandler({
      httpAgent: new Agent({
        /*params*/
      }),
    }),
    endpoint: "http://example.com",
  });
  ```

Reply:
Will do.